sql/src/test/java/io/crate/execution/engine/window/WindowFunctionTest.java

```diff
@@ -0,0 +1,168 @@
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.window;
+
+import com.google.common.collect.ImmutableMap;
+import io.crate.analyze.OrderBy;
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.auth.user.User;
+import io.crate.breaker.RamAccountingContext;
+import io.crate.data.InMemoryBatchIterator;
+import io.crate.data.Input;
+import io.crate.data.RowN;
+import io.crate.execution.engine.collect.InputCollectExpression;
+import io.crate.expression.InputFactory;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.Functions;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.doc.DocSchemaInfo;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.TestingTableInfo;
+import io.crate.sql.tree.QualifiedName;
+import io.crate.testing.SqlExpressions;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.elasticsearch.common.inject.AbstractModule;
+import org.hamcrest.Matcher;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static io.crate.data.SentinelRow.SENTINEL;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
+
+
+public class WindowFunctionTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private AbstractModule[] additionalModules;
+    private SqlExpressions sqlExpressions;
+    private Functions functions;
+    private TransactionContext txnCtx = CoordinatorTxnCtx.systemTransactionContext();
+    private InputFactory inputFactory;
+
+    private final RamAccountingContext RAM_ACCOUNTING_CONTEXT =
+        new RamAccountingContext("dummy", new NoopCircuitBreaker("dummy"));
+
+    public WindowFunctionTest(AbstractModule... additionalModules) {
+        this.additionalModules = additionalModules;
+    }
+
+    @Before
+    public void prepareFunctions() throws Exception {
+        DocTableInfo tableInfo = TestingTableInfo.builder(new RelationName(DocSchemaInfo.NAME, "users"), null)
+            .add("id", DataTypes.INTEGER)
+            .add("name", DataTypes.STRING)
```

Comment:
The tests for far onl yuse `a` and `x` - can we remove the other columns?

Reply:
shall we also rename the relation to`t1`as the fields have nothing to do with a possible `users` collection?

Reply:
done